newIDE/app/src/Profile/AuthenticatedUserProvider.js

```diff
@@ -323,17 +326,43 @@ export default class AuthenticatedUserProvider extends React.Component<
   };
 
   _fetchUserProfile = async (
-    options: ?{ dontNotifyAboutEmailVerification?: boolean }
+    options: ?{
+      dontNotifyAboutEmailVerification?: boolean,
+      resetState?: boolean,
+    }
   ) => {
     const { authentication } = this.props;
 
-    this.setState(({ authenticatedUser }) => ({
-      authenticatedUser: {
+    this.setState(({ authenticatedUser }) => {
+      let newAuthenticatedUser: AuthenticatedUser = {
         ...authenticatedUser,
         loginState: 'loggingIn',
         cloudProjectsFetchingErrorLabel: null,
-      },
-    }));
+      };
+      if (options && options.resetState) {
+        const {
+          cloudProjects,
+          receivedAssetPacks,
+          receivedAssetShortHeaders,
+          receivedGameTemplates,
+          badges,
+          notifications,
+        } = initialAuthenticatedUser;
```

Comment:
maybe all those props should be put in an object like `authenticatedUserLoadingAttributes` or similar, used in the `initialAuthenticatedUser`.
Just in case we add one in the future & forget to update it here